### PR TITLE
Enhance the contributing guide in online doc

### PR DESCRIPTION
**Pull request by nhuet**
Add sections and reorganize into:
- setting up dev environment
  - installation from source in dev mode
  - local build of the docs
  - running tests
- guide lines when coding
  - code linters
  - notebooks convention
  - adding tests
- pull request process

You can preview the resulting page [here](https://github.com/nhuet/scikit-decide/blob/nh/doc_complete_contribute_page/docs/contribute.md) (github markdown interpreting). Or have a complete view of resulting documentation by downloading the "doc" artifact from the "Build scikit-decide" action triggered by this PR , and then:
```shell
mkdir testdoc
cd testdoc
unzip path/to/doc.zip -d scikit-decide
python -m http.server
```

### CODE REVIEW

File: docs/contribute.md [hunk 1]
```diff
@@ -182,12 +241,60 @@ Can include a nice thumbnail.
...
+
+
+### Prior checks
+
+Before submitting your pull request, think to
+- [run the unit tests](#running-unit-tests)
+- [check the documentation locally](#building-the-docs-locally) if you modified it
+- check you respect the coding styles by [running linters](#coding-style-and-code-linting)
+
+I you do not, you will still be able to see the status of your PR as CI will do that checks for you.
```
Comment (dbarbier) [inline docs/contribute.md:286]: 2 typos: I**f** ... **these** checks for you

File: docs/contribute.md [hunk 2]
```diff
@@ -182,12 +241,60 @@ Can include a nice thumbnail.
...
- ```shell
- poetry run pytest -vv -s tests/solvers/cpp
+## Submitting pull requests
+
+When you think you are ready to merge your modifications into the main repository, you will have to open a pull request (PR).
+We can summarize the process as follows:
+
+- Fork the repository on github.
+- Clone your fork on your computer.
+- Create a branch with a self-explanatory branch, e.g. `xx/add-solver-crazyalgoname` where "xx" stands for your initials,
```
Comment (dbarbier) [inline docs/contribute.md:258]: External contributors do not need to follow this naming scheme, they can use their master branch if they want.

File: docs/contribute.md [hunk 3]
```diff
@@ -182,12 +241,60 @@ Can include a nice thumbnail.
...
+
+On your way to implement your contribution, you will probably have lots of commits,
+some modifying other ones from the same PR, or only modidying the code style.
+
+At the end of your work, consider reorganizing them by
+- squashing them into one or only a few logical commits
+- having a separate commit to reformat previous existing code if necessary
+- rewritting commit messages so that it explains the changes made and why, the "how" part being explained by the code itself
+  (see this [post](https://chris.beams.io/posts/git-commit/) about what a commit message should and should not contain)
+- rebasing on main repository main branch if it diverged too much by the time you finished
```
Comment (dbarbier) [inline docs/contribute.md:298]: upstream repository's master branch

**Review by dbarbier — APPROVED** [review]

File: docs/contribute.md (hunk 3 above)
Comment (nhuet) [reply docs/contribute.md]: done

File: docs/contribute.md (hunk 2 above)
Comment (nhuet) [reply docs/contribute.md]: done

File: docs/contribute.md (hunk 1 above)
Comment (nhuet) [reply docs/contribute.md]: done

**Pull request merged**